### PR TITLE
NU-659 define user trust level level

### PR DESCRIPTION
**Pull request by samykhalil-zz**
NU-659 define user trust level level depending on Moodle role

### CODE REVIEW

File: sso.php [hunk 1]
```diff
@@ -127,15 +127,26 @@
 // Get the user's description.
 $user = $DB->get_record('user', array('id' => $USER->id));
 $userdescription = format_text($user->description, $user->descriptionformat);
-
-$userlocale = explode('_', $USER->lang)[0];
+$group_lang = explode('_', $USER->lang)[0];
```
Comment (johntran777) [inline sso.php]: Refactor this into a method.

File: sso.php [hunk 2]
```diff
@@ -96,18 +96,58 @@
...
 // based on the email address.
 
+// get currunt user roles
+$trust_lvl= 1;
+global $USER, $DB;
+$usercontext = context_user::instance($USER->id);
+$roles=$DB->get_records_menu('role_assignments',array('userid'=>$USER->id),null,'id,roleid');
+$roles = array_unique($roles);
+
+if (in_array("5", $roles)) { //student roleid 5
```
Comment (johntran777) [inline sso.php]: Refactor the logic to get trust_lvl into a method.

**Review by johntran777 — COMMENTED** [review]
Comment.

File: sso.php [hunk 3]
```diff
@@ -96,18 +96,67 @@
...
 // Get the user's description.
+
 $user = $DB->get_record('user', array('id' => $USER->id));
 $userdescription = format_text($user->description, $user->descriptionformat);
-
-$userlocale = explode('_', $USER->lang)[0];
+function group_lang($user_lang){
+	$group_lang = explode('_', )[0];
+	return $group_lang;
+	}
```
Comment (alfonsocj) [inline sso.php]: Could you fix the indentation of this function

File: sso.php [hunk 4]
```diff
@@ -96,18 +96,67 @@
...
+//Match Discourse language code
+function user_locale($user_lang){
+switch ($user_lang) {
```
Comment (alfonsocj) [inline sso.php]: same here

File: sso.php [hunk 5]
```diff
@@ -96,18 +96,67 @@
...
 
 $extraparams = array(
     'username' => $USER->username,
     'name'     => fullname($USER, true),
     'bio'      => $userdescription,
-    'locale'   => $userlocale,
-    'groups' => 'lang_'.$userlocale
+    'locale'   => user_locale($USER->lang),
+    'groups' => 'lang_'.group_lang)$USER->lang),
+	'trust_level'=> user_role($USER->id)
```
Comment (alfonsocj) [inline sso.php]: here you are using a mix of tabs and spaces for indentation

**Review by alfonsocj — APPROVED** [review]

**Pull request merged**